tests/fakes/TheTest.php

```diff
@@ -40,4 +41,27 @@ public function testReturnsSubjectAfterApplyingContext(): void
             ))->value()
         );
     }
+
+    /**
+     * @covers ::__construct
+     * @covers ::value
+     * 
+     * @uses MaxGoryunov\SavingIterator\Fakes\SurveyEnvelope
+     * 
+     * @small
+     *
+     * @return void
+     */
+    public function testCallsContextOnSubject(): void
+    {
+        $name = "John";
+        $this->assertEquals(
+            $name,
+            (new The(
+                new stdClass(),
+                fn (stdClass $obj) => $obj->name = $name
+            ))
+                ->value()->name
+        );
```

Comment:
This test involves mutable state of stdClass. However, I do not know how to test this behaviour otherwise.